File: client/src/app/modules/tiers/services/tiers.category.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { TiersCategory } from '../../tiers/model/TiersCategory';

@Injectable({
  providedIn: 'root'
})
export class TiersCategoryService extends AppRestService<TiersCategory>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getTiersCategories() {
    return this.getListCached(new HttpParams(), "tiers-categories");
  }

  addOrUpdateTiersCategory(tiersCategory: TiersCategory) {
    this.clearListCache(new HttpParams(), "tiers-categories");
    return this.addOrUpdate(new HttpParams(), "tiers-category", tiersCategory, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/tiers/service/TiersCategoryService.java
package com.jss.osiris.modules.tiers.service;

import java.util.List;

import com.jss.osiris.modules.tiers.model.TiersCategory;

public interface TiersCategoryService {
    public List<TiersCategory> getTiersCategories();

    public TiersCategory getTiersCategory(Integer id);
	
	 public TiersCategory addOrUpdateTiersCategory(TiersCategory tiersCategory);
}
